### ISSUE TITLE
[ENH] Deprecate Differencer

### ISSUE THREAD

**Issue by TonyBagnall** [aeon-toolkit/aeon#1534]
### Describe the feature or idea you want to propose

The BaseTransformer class is simply a wrapper for this

```python
    for lag in lags:
        # converting lag to int since pandas complains if it's np.int64
        Xt = Xt.diff(periods=int(lag))
```
I guess it might be useful in a pipeline, but it does not work (see #700 ), the inverse is wrong

```
from aeon.transformations.difference import Differencer
from aeon.datasets import load_airline
y = load_airline()
transformer = Differencer(lags=[1, 12])
y_transform = transformer.fit_transform(y)
y_return = transformer.inverse_transform(y_transform)
# y_return not equal to y
```
and it does a strange padding operation.

 

### Describe your proposed solution

 I want to just remove it, can bring it back as a series transformer if necessary, but not implemented like this.
